compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -617,15 +617,40 @@ impl<'hir> LoweringContext<'_, 'hir> {
 
             hir::ExprKind::Closure(c)
         };
-        let generator = hir::Expr {
-            hir_id: self.lower_node_id(closure_node_id),
-            kind: generator_kind,
-            span: self.lower_span(span),
+        let parent_has_track_caller = self
+            .attrs
+            .values()
+            .find(|attrs| attrs.into_iter().find(|attr| attr.has_name(sym::track_caller)).is_some())
+            .is_some();
```

Comment:
I feel like this is very broken, the parent here might be just a normal function, not necessarily the desugaring of an `async fn`, see #105134